bob-apps/bin/bobd.rs

```diff
@@ -120,6 +146,81 @@ async fn main() {
     }
 }
 
+fn configure_testmode(sub_matches: &ArgMatches) -> AnyResult<(ClusterConfig, NodeConfig)> {
+    let mut addresses = Vec::with_capacity(1);
+    let port = match sub_matches.value_of("grpc-port") {
+        Some(v) => v.parse().context("could not parse --grpc-port")?,
+        None => 20000
+    };
+    let mut this_node = None;
+    if let Some(node_list) = sub_matches.value_of("nodes") {
+        let available_ips: Vec<String> = NetworkInterface::show()?.into_iter().filter_map(|itf|
+            match itf.addr? {
+                Addr::V4(addr) => {
+                    Some(addr.ip.to_string())
+                },
+                _ => None
+        }).collect();
+
+        for (index, addr) in node_list.split(",").enumerate() {
+            let split = &addr.split_once(":").context("could not find address in --nodes")?;
+            let in_ip = split.0;
+            let in_port = split.1.parse::<u16>().context("could not parse port in --nodes")?;
+            if this_node.is_none() {
+                this_node = available_ips.iter().find_map(|ip| {
+                    if ip == in_ip && port == in_port {
+                        Some(index)
+                    } else {
+                        None
+                    }
+                })
+            }
+            addresses.push(String::from(addr));
+        }
+        if this_node.is_none() {
```

Comment:
Below you have a line ` let this_node = this_node.unwrap();`. That line is a better place to convert to an error (you can use `context(..)` function)